olp-cpp-sdk-dataservice-read/src/VolatileLayerClientImpl.cpp

```diff
@@ -19,19 +19,107 @@
 
 #include "VolatileLayerClientImpl.h"
 
+#include <olp/core/client/CancellationContext.h>
+#include <olp/dataservice/read/PartitionsRequest.h>
+#include "repositories/DataRepository.h"
+#include "repositories/ExecuteOrSchedule.inl"
+#include "repositories/PartitionsRepository.h"
+
 namespace olp {
 namespace dataservice {
 namespace read {
 
+namespace {
+constexpr auto kLogTag = "VolatileLayerClientImpl";
+}
+
 VolatileLayerClientImpl::VolatileLayerClientImpl(
     olp::client::HRN catalog, std::string layer_id,
-    olp::client::OlpClientSettings client_settings) {}
+    olp::client::OlpClientSettings client_settings)
+    : catalog_(std::move(catalog)),
+      layer_id_(std::move(layer_id)),
+      client_settings_(std::move(client_settings)),
+      pending_requests_(std::make_shared<PendingRequests>()) {}
 
-VolatileLayerClientImpl::~VolatileLayerClientImpl() = default;
+VolatileLayerClientImpl::~VolatileLayerClientImpl() {
+  pending_requests_->CancelPendingRequests();
+}
 
 olp::client::CancellationToken VolatileLayerClientImpl::GetData(
     DataRequest data_request, Callback callback) {
-  return olp::client::CancellationToken{[]() {}};
+  auto catalog = catalog_;
+  auto client_settings = client_settings_;
+  auto layer_id = layer_id_;
+
+  auto get_data = [=](DataRequest data_request,
+                      olp::client::CancellationContext context,
+                      Callback callback) mutable {
```

Comment:
```suggestion
                      Callback callback) {
```
Should it be mutable?